SRVOCF-341: Add docs on developing TypeScript functions (#35587)
openshift/openshift-docs: modules/serverless-typescript-functions-context-objects.adoc

[id="serverless-typescript-functions-context-objects_{context}"]
= TypeScript context objects

> Functions are invoked by providing a `context` object as the first parameter.

Reviewer: ```suggestion
Functions are invoked with a `context` object as the first parameter.
```

> Functions are invoked by providing a `context` object as the first parameter.

Reviewer: ```suggestion
Functions are invoked with a `context` object as the first parameter.
```

.Example context object
[source,javascript]
----

> function handle(context, data)

Reviewer: ```suggestion
function handle(context:Context): string
```

export interface Context {
    log: Logger;
    req: IncomingMessage;
    query?: Record<string, any>;
    body?: Record<string, any>|string;
    method: string;
    headers: IncomingHttpHeaders;
    httpVersion: string;
    httpVersionMajor: number;
    httpVersionMinor: number;
    cloudevent: CloudEvent;
    cloudEventResponse(data: string|object): CloudEventResponse;
}

// CloudEventResponse is a convenience class used to create
// CloudEvents on function returns
export interface CloudEventResponse {
    id(id: string): CloudEventResponse;
    source(source: string): CloudEventResponse;
    type(type: string): CloudEventResponse;
    version(version: string): CloudEventResponse;
    response(): CloudEvent;
}
----

[id="serverless-typescript-functions-context-objects-cloudevent-data_{context}"]
== CloudEvent data

If the incoming request is a CloudEvent, any data associated with the CloudEvent is extracted from the event and provided as a second parameter. For example, if a CloudEvent is received that contains a JSON string in its data property that is similar to the following:

[source,json]
----
{
  "customerId": "0123456",
  "productId": "6543210"
}
----

> When invoked, the second parameter to the function, after the `context` object, will be a JavaScript object that has `customerId` and `productId` properties.

Author: @EricPonvelle I disagree, your suggested wording would imply that there are at least three parameters in total, and that the reference is to the third one.
